src/components/settings/Settings.js

```diff
@@ -0,0 +1,188 @@
+import React from 'react'
+import {
+  FlatList,
+  ScrollView,
+  View
+} from 'react-native'
+import PropTypes from 'prop-types';
+import { bindActionCreators } from 'redux'
+import EStyleSheet from 'react-native-extended-stylesheet'
+import { SettingsToggle, SettingHeader } from './settingsComponents'
+import NavBar from '../NavBar'
+import FontedText from '../common/FontedText'
+import Separator from '../common/Separator'
+import { connect } from 'react-redux'
+import {
+  checkPushPermissions,
+} from '../../actions/index'
+import GoogleAnalytics from 'react-native-google-analytics-bridge'
+import * as settingsActions from '../../actions/settings'
+
+GoogleAnalytics.trackEvent('view', 'Notification Settings')
+
+const mapStateToProps = (state) => {
+  const {
+    showAllWorkflows,
+    enableNotifications,
+    newProjectNotifications,
+    newBetaNotifications,
+    urgentHelpNotification,
+    projectSpecificNotifications
+  } = state.settings
+
+  return {
+    notifications: state.main.notifications,
+    projectList: state.projects.projectList || [],
+    isFetching: state.main.isFetching,
+    errorMessage: state.main.errorMessage,
+    pushEnabled: state.main.pushEnabled,
+    showAllWorkflows,
+    enableNotifications,
+    newProjectNotifications,
+    newBetaNotifications,
+    urgentHelpNotification,
+    projectSpecificNotifications
+  }
+}
+
+const mapDispatchToProps = (dispatch) => ({
+  checkPushPermissions(){
+    dispatch(checkPushPermissions())
+  },
+  settingsActions: bindActionCreators(settingsActions, dispatch)
+})
+
+export class Settings extends React.Component {
+  componentDidMount() {
+    this.props.checkPushPermissions()
+  }
+
+  static renderNavigationBar() {
+    return <NavBar title={'Settings'} showBack={true} />;
+  }
+
+  renderSettingsToggle = ({item}) => {
+    const {subscribed, id, display_name } = item
+    return (
+      <SettingsToggle
+        value={subscribed}
+        onToggle={() => {this.props.settingsActions.updateProjectSubsciption(id, !subscribed)}}
+        title={display_name}
+        disabled={!this.props.enableNotifications}
+      />
+    )
+  }
+
+  render() {
+    
+
+    return (
+      <ScrollView contentContainerStyle={styles.scrollViewContainer}>
+        <SettingHeader text="General Settings" />
+        <View style={styles.workflowContainer}>
+          <SettingsToggle
+            title="Show all workflows"
+            description="Includes non-native workflows"
+            onToggle={() => {this.props.settingsActions.updateShowAllWorkflow(!this.props.showAllWorkflows)}}
+            value={this.props.showAllWorkflows}
+          />
+        </View>
+        <Separator style={styles.titlePadding} color={'rgba(151, 151, 151, 1)'}/>
```

Comment:
Can this be given a global value or constant? Not sure what 151, 151, 151, 1 converts to.